### PR TITLE
RavenDB-10923: Rename CreateTemporaryBufferPager if cannot be created…

### PR DESCRIPTION
**Pull request by aviviadi**
… due to ENOENT although exists

### CODE REVIEW

File: src/Voron/StorageEnvironmentOptions.cs [hunk 1]
```diff
@@ -675,18 +676,19 @@ void Rename()
...
             // This is used for special pagers that are used as temp buffers and don't 
             // require encryption: compression, recovery, lazyTxBuffer.
             public override AbstractPager CreateTemporaryBufferPager(string name, long initialSize)
             {
-                var scratchFile = TempPath.Combine(name);
-                if (File.Exists(scratchFile.FullPath))
-                    File.Delete(scratchFile.FullPath);
-
-                var pager = GetMemoryMapPagerInternal(this, initialSize, scratchFile, deleteOnClose: true);
+                var pager = GetMemoryMapPagerWithPossibleFileRename(name, initialSize, GetMemoryMapPagerInternal);
```
Comment (ayende) [inline src/Voron/StorageEnvironmentOptions.cs:691]: Given that they are so similar, is there a reason not to just call CreateScratchPager here and avoid `GetMemoryMapPagerWithPossibleFileRename` ?

Comment (ayende) [reply src/Voron/StorageEnvironmentOptions.cs:691]: going to implement this in this manner.

Comment (aviviadi) [reply src/Voron/StorageEnvironmentOptions.cs:691]: https://github.com/ravendb/ravendb/pull/6648

Comment (ayende) [reply src/Voron/StorageEnvironmentOptions.cs:691]: Wasn't aware of that, good point

**Pull request closed without merging**